JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/movies/MovieDetailsScreenViewModel.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jetstream.presentation.screens.movies
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.createSavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.CreationExtras
+import com.google.jetstream.data.entities.MovieDetails
+import com.google.jetstream.data.repositories.MovieRepository
+import com.google.jetstream.data.repositories.MovieRepositoryImpl
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.stateIn
+
+class MovieDetailsScreenViewModel(
+    savedStateHandle: SavedStateHandle,
+    repository: MovieRepository = MovieRepositoryImpl.instance,
+) : ViewModel() {
+
+    val id: String = checkNotNull(savedStateHandle[MovieDetailsScreen.MovieIdBundleKey])
```

Comment:
How about this?
```
val uiState: StateFlow<LoadingState> = savedStateHandle.getStateFlow<String?>("MovieDetailsScreen.MovieIdBundleKey").flatMapLatest { id ->
    if(id == null) {
       flowOf(LoadingState.Error)
    } else {
        repository.getMovieDetails(movieId = id).map(LoadingState::Done)
    }
}
```

This way we graciously handle a situation where the id is null.